### PR TITLE
Implement Home page

### LINKED ISSUE #217: Implement Home page
As of creation of this ticket, the web app have a very minimal Home page saying "Coming Soon" as seen on the demo deployment: https://demo.verida.one

In a separate deployment (with WebFlow), we have a landing page at https://www.verida.one for marketing purpose.
Ultimately, the web app will be hosted on ww.verida.one and the WebFlow landing page will be decommissioned. Meaning the web app must have a (useful) Home page.

For now, we can "basically" replicate what's the landing page as a Home page.

Except maybe for the header and footer which components have already been implemented.

### PR DESCRIPTION
**Pull request by icetrust0212**

### CODE REVIEW

**Comment by aurelticot** [comment]
Thanks @icetrust0212 I'll review it today

File: src/components/templates/AppLayout.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ export const AppLayout: React.FunctionComponent = () => {
             }`}
           >
             <HeaderOffset />
-            <main className="mx-auto flex w-full max-w-screen-sm flex-grow flex-col px-4 pt-4">
+            <main className="mx-auto flex w-full max-w-[1248px] flex-grow flex-col px-4 pt-4">
```
Comment (aurelticot) [inline src/components/templates/AppLayout.tsx]: This actually impacts all the other pages of the app. They have a max width on the `sm` breakpoint as per our design.
If the Home page intends to be wider, and so need a different value here, we could identify it with its `location.pathname === "/"` and adapt the class name accordingly

File: tailwind.config.js [hunk 2]
```diff
@@ -23,13 +23,19 @@ module.exports = {
...
     },
     extend: {
       screens: {
-        sm: "696px",
+        '2xl': { max: '1800px' },
+        xl: { max: '1535px' },
+        lg: { max: '1279px' },
+        md: { max: '992px' },
+        sm: { max: '696px' },
+        xs: { max: '479px' },
```
Comment (aurelticot) [inline tailwind.config.js]: What is the justification of redefining the breakpoints with `max`?

File: src/components/pages/index.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 export * from "./AssetDetailsView";
 export * from "./BadgeListView";
 export * from "./CollectibleListView";
-export * from "./HomeView";
+export * from "./HomeView/HomeView";
```
Comment (aurelticot) [inline src/components/pages/index.ts]: Could you add a barrel `index.ts` file under HomeView and export what's needed? Thanks

Comment (aurelticot) [reply src/components/pages/index.ts]: Worth pointing this repository uses an atomic design structure in which the components used by `HomeView` would have normally be located in the atoms, molecules or organisms folders

File: package.json [hunk 4]
```diff
@@ -56,6 +57,7 @@
     "eslint-plugin-promise": "^6.0.0",
     "eslint-plugin-react": "^7.34.0",
     "eslint-plugin-react-hooks": "^4.6.0",
+    "framer-motion": "^11.0.14",
```
Comment (aurelticot) [inline package.json:60]: 👍

File: src/components/organisms/Footer.tsx [hunk 5]
```diff
@@ -55,28 +58,23 @@ export const Footer: React.FunctionComponent = () => {
...
+          {documentationLinkLabel}
+        </a>
+        <a
+          href={VERIDA_TERMS_AND_CONDITIONS_URL}
+          target="_blank"
+          rel="noopener noreferrer"
+          className="hover:underline"
+        >
+          {termsAndConditionLinkLabel}
+        </a>
```
Comment (aurelticot) [inline src/components/organisms/Footer.tsx]: Not quite sure of the justification to modify the footer.
Anyway, on wide window the "Built on" is not centered anymore.
And on narrow window, the gap between the elements is not homogeneous (see "Documentation" and "Terms & Conditions").

Maybe better to revert these changes.

Comment (aurelticot) [reply src/components/organisms/Footer.tsx]: Screenshot 2024-03-20 at 3 50 02 pm

Screenshot 2024-03-20 at 3 49 45 pm

File: src/components/pages/HomeView/HomeView.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+import { useIntl } from "react-intl";
+import HeroSection from "./HeroSection";
+import JoinWaitlist from "./JoinWaitlist";
+import LearnMore from "./LearnMore";
+import Showcase from "./Showcase";
+
+export const HomeView: React.FunctionComponent = () => {
+  const i18n = useIntl();
+
+  const comingSoonMessage = i18n.formatMessage({
```
Comment (aurelticot) [inline src/components/pages/HomeView/HomeView.tsx]: Should probably get a more relavant message for image alt

File: src/components/pages/HomeView/HeroSection.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+      <div className="mx-auto overflow-hidden rounded-xl border-[3px] border-white/50">
+        <button className="mx-auto bg-[#efefef] px-10 py-3 text-base font-semibold text-background xs:px-4">
+          {joinTheWaitlistButtonLabel}
+        </button>
+      </div>
+    </motion.div>
+  );
+};
+
+export default HeroSection;
```
Comment (aurelticot) [inline src/components/pages/HomeView/HeroSection.tsx]: We use named exports in this repository if you could update this and the other component. Thanks

File: src/components/pages/HomeView/HeroSection.tsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
+import { useIntl } from "react-intl";
+import { motion } from "framer-motion";
+
+const HeroSection = () => {
+  const i18n = useIntl();
+
+  const heroTagLabel = i18n.formatMessage({
+    id: "HomeView.Tag",
```
Comment (aurelticot) [inline src/components/pages/HomeView/HeroSection.tsx]: The id of localised messages should follow the convention `<ComponentName>.<variableName>`. You can refer the README for more info.
I won't make a comment on the other messages but they should be updated as well.

File: src/components/pages/HomeView/HomeView.tsx [hunk 9]
```diff
@@ -0,0 +1,41 @@
+import React from "react";
+// import { ReactComponent as VeridaOneLogo } from "assets/images/verida_one_logo_with_text.svg";
```
Comment (aurelticot) [inline src/components/pages/HomeView/HomeView.tsx]: Could you delete commented code? Thanks

File: src/components/pages/HomeView/JoinWaitlist.tsx [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+          y: 80,
+          opacity: 0,
+        },
+        visible: {
+          y: 0,
+          opacity: 100,
+        },
+      }}
+      transition={{ duration: 0.8 }}
+      className="!mt-40 flex flex-col"
```
Comment (aurelticot) [inline src/components/pages/HomeView/JoinWaitlist.tsx]: This margin should not be applied by the component root tag as it's not related to the style/layout of the component itself. It sshould be applied by whatever component is rendering it (HomeView in that case). The HomeView should either surround this component with a div and add the margin there. Or the props of this JoinWaitlist component should take the style and merge it with there in the root tag, so that HomeView apply the style directly on the component. (Even better, the props of JoinWaitlist should extend `ComponentProps<'div'>` and be passed to the root div tag, that's actually what I tend to do personally but I notice I haven't really done it in this repo)

File: src/components/pages/HomeView/LearnMore.tsx [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+        visible: {
+          y: 0,
+          opacity: 100,
+        },
+      }}
+      transition={{ duration: 0.8 }}
+      className="grid grid-cols-2 gap-8 py-40 md:grid-cols-1"
+    >
+      <ContentBox
+        title="Developer-Friendly"
```
Comment (aurelticot) [inline src/components/pages/HomeView/LearnMore.tsx]: Should be a localised message. Same as for the description.
And same for the other box "Decentralized"

File: src/components/organisms/Header.tsx [hunk 12]
```diff
@@ -55,20 +55,21 @@ export const Header: React.FC = () => {
...
+      className={`grid grid-cols-[minmax(165px,_1fr)_minmax(min-content,696px)_minmax(165px,_1fr)] border-b border-solid border-gray-dark bg-background/10 px-4 py-4 backdrop-blur-[10px] sm:flex sm:gap-4 sm:px-6`}
     >
       <h1 className={`col-start-1 justify-self-start ${hideWhenSearch}`}>
         <Link to="/">
-          <div className="aspect-[10/6.97] h-10 sm:hidden">
+          <div className="hidden aspect-[10/6.97] h-10 sm:block">
             <VeridaOneLogo height="100%" width="100%" />
           </div>
-          <div className="hidden aspect-[10/3] h-10 sm:block">
+          <div className="aspect-[10/3] h-10 sm:hidden">
```
Comment (aurelticot) [inline src/components/organisms/Header.tsx]: Not quite sure of the justification of these changes on the header. Yes, the one on https://verida.one is different than the one on https://demo.verida.one but I specified in the ticket to not change it.

That being said we can find a middle-ground for the opacity. Either be a value working on every page, or in some way applying `10` on the Home page and `80` on the others.

Anyway, the layout is broken!
The narrow layout is applied to wide windows and the wide layout is applied to narrow windows. The Search is particularly impacted.

The positioning with the SearchResult is also impacted.

Again, I think it's better to revert to how it was.

File: public/Star.png
Comment (aurelticot) [inline public/Star.png:1]: Could you move all the assets under a folder rather than at the root of `public`? Just for a better organisation. Thanks

**Review by aurelticot — CHANGES_REQUESTED** [review]
Thanks for the PR @icetrust0212, really appreciate the work.

There are a few changes needed though, notably the Header is not behaving correctly anymore.
Please see my comments below.

File: src/components/templates/AppLayout.tsx (hunk 1 above)
Comment (icetrust0212) [reply src/components/templates/AppLayout.tsx]: resolved

File: tailwind.config.js (hunk 2 above)
Comment (icetrust0212) [reply tailwind.config.js]: I updated it to the previous one

File: src/components/pages/index.ts (hunk 3 above)
Comment (icetrust0212) [reply src/components/pages/index.ts]: Resolved

File: src/components/organisms/Footer.tsx (hunk 5 above)
Comment (icetrust0212) [reply src/components/organisms/Footer.tsx]: I reverted the changes

File: src/components/pages/HomeView/HomeView.tsx (hunk 9 above)
Comment (icetrust0212) [reply src/components/pages/HomeView/HomeView.tsx]: Resolved, removed the commented code

File: src/components/pages/HomeView/JoinWaitlist.tsx (hunk 10 above)
Comment (icetrust0212) [reply src/components/pages/HomeView/JoinWaitlist.tsx]: Resolved

File: src/components/pages/HomeView/LearnMore.tsx (hunk 11 above)
Comment (icetrust0212) [reply src/components/pages/HomeView/LearnMore.tsx]: Resolved

File: src/components/organisms/Header.tsx (hunk 12 above)
Comment (icetrust0212) [reply src/components/organisms/Header.tsx]: I reverted the changes

File: public/Star.png
Comment (icetrust0212) [reply public/Star.png:1]: Resolved, I moved all the assets from public folder to src/assets folder

File: src/components/molecules/ContentBox.tsx [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+        src={mainImageUrl}
+        alt={``}
+        className={cx("relative z-[10] mx-auto max-w-full", mainImageClassName)}
+      />
+      {children}
+    </motion.div>
+  );
+};
+
+export default ContentBox;
```
Comment (aurelticot) [inline src/components/molecules/ContentBox.tsx]: We use named exports in this repository. Thanks

File: src/components/organisms/HeroSection.tsx [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+      <div className="mx-auto overflow-hidden rounded-xl border-[3px] border-white/50">
+        <button className="mx-auto bg-[#efefef] px-4 py-3 text-base font-semibold text-background sm:px-10">
+          {joinTheWaitlistButtonLabel}
+        </button>
+      </div>
+    </motion.div>
+  );
+};
+
+export default HeroSection;
```
Comment (aurelticot) [inline src/components/organisms/HeroSection.tsx]: We use named exports in this repository. Thanks

File: src/components/organisms/JoinWaitlist.tsx [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+      <div className="mx-auto overflow-hidden rounded-xl border-[3px] border-white/50">
+        <button className="mx-auto bg-[#efefef] px-4 py-3 text-base font-semibold text-background sm:px-10 ">
+          {joinTheWaitlistButtonLabel}
+        </button>
+      </div>
+    </motion.div>
+  );
+};
+
+export default JoinWaitlist;
```
Comment (aurelticot) [inline src/components/organisms/JoinWaitlist.tsx]: We use named exports in this repository. Thanks

File: src/components/organisms/LearnMore.tsx [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+        >
+          <span className="text-base font-semibold">{learnMore}</span>
+          <img src={ArrowLeftIcon} alt={``} className="brightness-[1000]" />
+        </a>
+      </ContentBox>
+    </motion.div>
+  );
+};
+
+export default LearnMore;
```
Comment (aurelticot) [inline src/components/organisms/LearnMore.tsx]: We use named exports in this repository. Thanks

File: src/components/organisms/Showcase.tsx [hunk 17]
```diff
@@ -0,0 +1,250 @@
...
+              willChange: "transform",
+            }}
+          />
+        </ContentBox>
+      </div>
+    </>
+  );
+};
+
+export default Showcase;
```
Comment (aurelticot) [inline src/components/organisms/Showcase.tsx]: We use named exports in this repository. Thanks

File: src/components/organisms/LearnMore.tsx [hunk 18]
```diff
@@ -0,0 +1,98 @@
+import { useIntl } from "react-intl";
+import { motion } from "framer-motion";
+import ContentBox from "../molecules/ContentBox";
```
Comment (aurelticot) [inline src/components/organisms/LearnMore.tsx]: We use barrel files in this repository. Could you add this component to the barrel file in `molecules`? Thanks

File: src/components/organisms/Showcase.tsx [hunk 19]
```diff
@@ -0,0 +1,250 @@
+import { useIntl } from "react-intl";
+import ContentBox from "../molecules/ContentBox";
```
Comment (aurelticot) [inline src/components/organisms/Showcase.tsx]: We use barrel files in this repository. Could you add this component to the barrel file in `molecules`? Thanks

File: src/components/pages/HomeView.tsx [hunk 20]
```diff
@@ -1,6 +1,11 @@
 import React from "react";
-import { ReactComponent as VeridaOneLogo } from "assets/images/verida_one_logo_with_text.svg";
 import { useIntl } from "react-intl";
+import HeroSection from "../organisms/HeroSection";
+import JoinWaitlist from "../organisms/JoinWaitlist";
+import LearnMore from "../organisms/LearnMore";
+import Showcase from "../organisms/Showcase";
```
Comment (aurelticot) [inline src/components/pages/HomeView.tsx]: We use barrel files in this repository. Could you add this component to the barrel file in `organisms`? Thanks

File: src/components/pages/HomeView.tsx [hunk 21]
```diff
@@ -1,6 +1,11 @@
 import React from "react";
-import { ReactComponent as VeridaOneLogo } from "assets/images/verida_one_logo_with_text.svg";
 import { useIntl } from "react-intl";
+import HeroSection from "../organisms/HeroSection";
+import JoinWaitlist from "../organisms/JoinWaitlist";
+import LearnMore from "../organisms/LearnMore";
```
Comment (aurelticot) [inline src/components/pages/HomeView.tsx]: We use barrel files in this repository. Could you add this component to the barrel file in `organisms`? Thanks

File: src/components/pages/HomeView.tsx [hunk 22]
```diff
@@ -1,6 +1,11 @@
 import React from "react";
-import { ReactComponent as VeridaOneLogo } from "assets/images/verida_one_logo_with_text.svg";
 import { useIntl } from "react-intl";
+import HeroSection from "../organisms/HeroSection";
+import JoinWaitlist from "../organisms/JoinWaitlist";
```
Comment (aurelticot) [inline src/components/pages/HomeView.tsx]: We use barrel files in this repository. Could you add this component to the barrel file in `organisms`? Thanks

File: src/components/pages/HomeView.tsx [hunk 23]
```diff
@@ -1,6 +1,11 @@
 import React from "react";
-import { ReactComponent as VeridaOneLogo } from "assets/images/verida_one_logo_with_text.svg";
 import { useIntl } from "react-intl";
+import HeroSection from "../organisms/HeroSection";
```
Comment (aurelticot) [inline src/components/pages/HomeView.tsx]: We use barrel files in this repository. Could you add this component to the barrel file in `organisms`? Thanks

File: src/components/pages/HomeView.tsx [hunk 24]
```diff
@@ -12,11 +17,28 @@ export const HomeView: React.FunctionComponent = () => {
...
+      <div className="mt-40">
+        <JoinWaitlist />
       </div>
-      <p className="text-xl md:text-4xl">{comingSoonMessage}</p>
+
+      <LearnMore />
+
+      <img
+        src={FooterBg}
+        alt={comingSoonMessage}
```
Comment (aurelticot) [inline src/components/pages/HomeView.tsx]: This alt message is not relevant for this image. Can you update it please? Thanks

File: src/components/pages/HomeView.tsx [hunk 25]
```diff
@@ -12,11 +17,28 @@ export const HomeView: React.FunctionComponent = () => {
   });
 
   return (
-    <div className="flex flex-grow flex-col items-center justify-center space-y-12 p-8">
-      <div className="aspect-[10/3] w-full">
-        <VeridaOneLogo height="100%" width="100%" />
+    <div className="flex flex-grow flex-col items-center justify-center space-y-12">
+      <img
+        src={HeroBg}
+        alt={comingSoonMessage}
```
Comment (aurelticot) [inline src/components/pages/HomeView.tsx]: This alt message is not relevant for this image. Can you update it please? Thanks

**Review by aurelticot — CHANGES_REQUESTED** [review]
All working fine. Thanks.

Just added a few comments for changes to go with this repository's style.

File: src/components/molecules/ContentBox.tsx (hunk 13 above)
Comment (icetrust0212) [reply src/components/molecules/ContentBox.tsx]: Done
`export const ContentBox: React.FC<ContentBoxProps> = (props) => {...}`

File: src/components/organisms/HeroSection.tsx (hunk 14 above)
Comment (icetrust0212) [reply src/components/organisms/HeroSection.tsx]: Done

File: src/components/organisms/JoinWaitlist.tsx (hunk 15 above)
Comment (icetrust0212) [reply src/components/organisms/JoinWaitlist.tsx]: Done

File: src/components/organisms/LearnMore.tsx (hunk 16 above)
Comment (icetrust0212) [reply src/components/organisms/LearnMore.tsx]: Done

File: src/components/organisms/Showcase.tsx (hunk 17 above)
Comment (icetrust0212) [reply src/components/organisms/Showcase.tsx]: Done

File: src/components/organisms/LearnMore.tsx (hunk 18 above)
Comment (icetrust0212) [reply src/components/organisms/LearnMore.tsx]: Done

File: src/components/organisms/Showcase.tsx (hunk 19 above)
Comment (icetrust0212) [reply src/components/organisms/Showcase.tsx]: Done

File: src/components/pages/HomeView.tsx (hunk 20 above)
Comment (icetrust0212) [reply src/components/pages/HomeView.tsx]: Done

File: src/components/pages/HomeView.tsx (hunk 21 above)
Comment (icetrust0212) [reply src/components/pages/HomeView.tsx]: Done

File: src/components/pages/HomeView.tsx (hunk 22 above)
Comment (icetrust0212) [reply src/components/pages/HomeView.tsx]: Done

File: src/components/pages/HomeView.tsx (hunk 23 above)
Comment (icetrust0212) [reply src/components/pages/HomeView.tsx]: Done

File: src/components/pages/HomeView.tsx (hunk 24 above)
Comment (icetrust0212) [reply src/components/pages/HomeView.tsx]: Done

File: src/components/pages/HomeView.tsx (hunk 25 above)
Comment (icetrust0212) [reply src/components/pages/HomeView.tsx]: Done

**Review by aurelticot — APPROVED** [review]
Thanks @icetrust0212
